koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -396,25 +397,44 @@ def _write_logs(self, build_id, logs_dir):
 
         user_warnings = UserWarningsStore()
 
-        with open(log_filename, 'wb') as outfile:
-            for line in logs:
-                if METADATA_TAG in line:
-                    _, meta_file = line.rsplit(' ', 1)
-                    source_file = os.path.join(koji.pathinfo.work(), meta_file)
-                    uploadpath = os.path.join(logs_dir, os.path.basename(meta_file))
-                    shutil.copy(source_file, uploadpath)
-                    continue
+        for task_run_name, line in logs:
+            if METADATA_TAG in line:
+                _, meta_file = line.rsplit(' ', 1)
+                source_file = os.path.join(koji.pathinfo.work(), meta_file)
+                uploadpath = os.path.join(logs_dir, os.path.basename(meta_file))
+                shutil.copy(source_file, uploadpath)
+                continue
+
+            if user_warnings.is_user_warning(line):
+                user_warnings.store(line)
+                continue
+
+            if platforms:
+                task_platform = next(
+                    (platform for platform in platforms if
+                     platform.replace('_', '-') in task_run_name),
+                    'noarch'
+                )
+            else:
+                task_platform = 'noarch'
 
-                if user_warnings.is_user_warning(line):
-                    user_warnings.store(line)
-                    continue
+            if task_platform not in logfiles:
+                log_filename = self.incremental_log_basename.replace('osbs-build',
+                                                                     f'osbs-build-{task_platform}')
```

Comment:
For reference, this is how OSBS 1 handles platform logs: https://github.com/containerbuildsystem/koji-containerbuild/blob/c1201d7551483e7459c08fd1098ecb27e9f3ef77/koji_containerbuild/plugins/builder_containerbuild.py#L400

Notably, the log filenames for the per-arch logs should be just "{platform}.log" (like in atomic-reactor)